cstore_reader.c

```diff
@@ -1101,22 +1129,26 @@ DeserializeDatumArray(StringInfo datumBuffer, bool *existsArray, uint32 datumCou
  * DeserializeBlockData deserializes requested data block for all columns and
  * stores in blockDataArray. It uncompresses serialized data if necessary. The
  * function also deallocates data buffers used for previous block, and compressed
- * data buffers for the current block which will not be needed again.
+ * data buffers for the current block which will not be needed again. If a column
+ * data is not present serialized buffer, than default value (or null) is used
+ * to fill value array.
  */
 static void
 DeserializeBlockData(StripeBuffers *stripeBuffers, uint64 blockIndex,
 					 Form_pg_attribute *attributeFormArray, uint32 rowCount,
-					 ColumnBlockData **blockDataArray)
+					 ColumnBlockData **blockDataArray, TupleDesc tupleDescriptor)
 {
 	int columnIndex = 0;
 	for (columnIndex = 0; columnIndex < stripeBuffers->columnCount; columnIndex++)
 	{
+		ColumnBlockData *blockData = blockDataArray[columnIndex];
+		Form_pg_attribute attributeForm = attributeFormArray[columnIndex];
 		ColumnBuffers *columnBuffers = stripeBuffers->columnBuffersArray[columnIndex];
+		bool columnAdded = (columnBuffers == NULL) && (blockData != NULL);
```

Comment:
When I wrote similar boolean assignment earlier, Ozgun suggested to use the following form instead:

```C
bool columnAdded = false;
if ((columnBuffers == NULL) && (blockData != NULL))
{
	columnAdded = true;
}